### PR TITLE
Provide selectable page size controls for RSDs

### PR DESCRIPTION
**Pull request by wgu-edwin**
[Java CI with Maven](https://github.com/wgu-opensource/osmt/actions/workflows/maven.yml)
Created dropdown for choosing the amount of RSDs to display per page

sizePagination

### CODE REVIEW

File: ui/src/app/collection/detail/collection-public/collection-public.component.ts [hunk 1]
```diff
@@ -26,7 +27,7 @@ export class CollectionPublicComponent extends Whitelabelled implements OnInit {
   results: PaginatedSkills | undefined
 
   from = 0
-  size = 50
+  size = this.pageSizeOptions?.length > 0 ? this.pageSizeOptions[0] : 50
```
Comment (JohnKallies) [inline ui/src/app/collection/detail/collection-public/collection-public.component.ts:30]: Can pageSizeOptions be a constant? We then don't have to null check it.

File: ui/src/app/richskill/list/skills-list.component.ts [hunk 2]
```diff
@@ -25,8 +25,9 @@ import {FilterDropdown} from "../../models/filter-dropdown.model"
 })
 export class SkillsListComponent extends QuickLinksHelper {
 
+  pageSizeOptions = [50, 100, 150]
```
Comment (JohnKallies) [inline ui/src/app/richskill/list/skills-list.component.ts:28]: again, as a constant?

File: ui/src/app/collection/detail/collection-public/collection-public.component.ts (hunk 1 above)
Comment (manuel-delvillar) [reply ui/src/app/collection/detail/collection-public/collection-public.component.ts:30]: Do you mean something like this?
```
  pageSizeOptions = [50, 100, 150]
  from = 0
  size = 50
```

Or maybe a constant from `size-pagination.component.ts
`

`values: number[] = [50, 100, 150]`

Comment (JohnKallies) [reply ui/src/app/collection/detail/collection-public/collection-public.component.ts]: const pageSizeOptions = [50, 100, 150]
...
  // initialize size, no null check, and reassign it in the handler 
  size = this.pageSizeOptions[0]

File: ui/src/app/table/skills-library-table/size-pagination/size-pagination.component.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import {FormControl} from "@angular/forms"
+
+@Component({
+  selector: "app-size-pagination",
+  templateUrl: "./size-pagination.component.html",
+  styleUrls: ["./size-pagination.component.scss"]
+})
+export class SizePaginationComponent implements OnInit {
+
+  values: number[] = [50, 100, 150]
```
Comment (Corpratespaz) [inline ui/src/app/table/skills-library-table/size-pagination/size-pagination.component.ts:11]: Technically, this should be read only:
`readonly values: number[] = [50, 100, 150]`
if not typed as an enumeration. Similarly, if not an enum the input to onValueChanged() should be checked/validated against the values array. As it stands you can inject values from the browser to paginate on any number through this new component. Just an FYI, not required to change it for this feature but important to be aware of in general.

Comment (wgu-edwin) [reply ui/src/app/table/skills-library-table/size-pagination/size-pagination.component.ts]: readonly has been added, thanks David

File: ui/src/app/app.module.ts [hunk 4]
```diff
@@ -210,7 +212,8 @@ export function initializeApp(
     ReactiveFormsModule,
     CommonModule,
     SharedModule,
-    FormModule
+    FormModule,
```
Comment (JohnKallies) [inline ui/src/app/app.module.ts:215]: Can we please alias either FormModule (as OsmtFormModule) or FormsModule (as NgFormsModule)

**Review by jchavez137 — APPROVED** [review]

**Pull request merged**